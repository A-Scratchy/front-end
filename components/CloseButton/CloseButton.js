import classNames from 'classnames';
import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
import { bool, func, oneOf } from 'prop-types';
import React, { Component } from 'react';
import PlusIcon from 'static/images/icons/plus.svg';
import styles from './CloseButton.css';

export default class CloseButton extends Component {
  static propTypes = {
    disabled: bool,
    onClick: func.isRequired,
    theme: oneOf(['primary', 'secondary', 'white']),
  };

  static defaultProps = {
    disabled: false,
    theme: 'secondary',
  };

  render() {
    const { disabled, onClick, theme } = this.props;
    return (
      <button className={styles.CloseButton} disabled={disabled} onClick={onClick} type="button">
        <ScreenReaderOnly>Close</ScreenReaderOnly>

        <PlusIcon className={classNames(styles.icon, styles[theme])} />
      </button>
    );
  }
}
