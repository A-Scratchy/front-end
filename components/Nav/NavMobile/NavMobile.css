.NavMobile {
  display: flex;
  position: absolute;
  background-color: var(--gray);
  height: var(--navBarHeight);
  width: 100%;
}

.logo {
  position: relative;
  max-height: 50%;
  max-width: 50%;
  top: 16.25px;
  left: 2rem;
  cursor: pointer;
}

.hamburgerButtonWrapper {
  position: absolute;
  width: 35px;
  top: 16px;
  right: 2rem;
  cursor: pointer;
  padding: 0;
  border: none;
  background: none;
}

.hamburgerIcon {
  width: 35px;
}

.checkButtonWrapper {
  position: absolute;
  top: 10px;
  right: 10px;
  cursor: pointer;
  padding: 0;
  border: none;
  background: none;
}

.checkIcon {
  width: 35px;
}

.dropdown {
  position: absolute;
  left: 0;
  background: rgba(37, 46, 62, 0.97);
  width: 95%;
  height: auto;
  z-index: 100;
}

.ul {
  padding: 15px;
  margin: 0;
  list-style: none;
  font-size: 14px;
}

.li {
  padding: 10px;
}

.li:first-child {
  padding-top: 30px;
}

.li:last-child {
  padding-bottom: 40px;
}

.a {
  text-decoration: none;
  color: var(--white);
  fill: var(--white);
  transition: all 0.2s linear;
  font-weight: bolder;
  font-size: 1.7rem;
  display: inline-block;
  height: 34px;
}

.a:hover {
  cursor: pointer;
  color: var(--primary);
  fill: var(--primary);
  transition: all 0.2s linear;
}
