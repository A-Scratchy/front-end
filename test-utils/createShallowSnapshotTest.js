/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
