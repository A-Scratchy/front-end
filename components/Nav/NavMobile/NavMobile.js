import React, { Component } from 'react';
import Link from 'next/link';
import { donateLink, s3 } from 'common/constants/urls';
import { navItems } from 'common/constants/navigation';
import flattenDepth from 'lodash/flattenDepth';
import HamburgerIcon from 'static/images/icons/hamburger.svg';
import CloseButton from 'components/_common_/CloseButton/CloseButton';
import styles from './NavMobile.css';

export default class NavMobile extends Component {
  state = {
    isDropDownVisible: false,
  };

  toggleDropdown = () =>
    this.setState(prevState => ({ isDropDownVisible: !prevState.isDropDownVisible }));

  handleOnClick = () => this.toggleDropdown();

  render() {
    const { state } = this;
    const { isDropDownVisible } = state;
    const links = flattenDepth(navItems.map(navItem => [navItem, navItem.sublinks]), 2);

    return (
      <header className={styles.NavMobile}>
        <Link href="/">
          <img
            src={`${s3}branding/logos/small-blue-logo.png`}
            alt="Operation Code Logo"
            className={styles.logo}
          />
        </Link>

        <button
          className={styles.hamburgerButtonWrapper}
          type="button"
          name="dropdown"
          onClick={this.handleOnClick}
        >
          <HamburgerIcon className={styles.hamburgerIcon} />
        </button>

        {isDropDownVisible && (
          <div className={styles.dropdown}>
            <CloseButton onClick={this.handleOnClick} theme="white" />
            <ul className={styles.ul}>
              <li className={styles.li}>
                <Link href="/">
                  <a className={styles.a}>Home</a>
                </Link>
              </li>
              {links.map(navlink => (
                <li className={styles.li} key={navlink.name}>
                  <Link href={navlink.href}>
                    <a className={styles.a}>{navlink.name}</a>
                  </Link>
                </li>
              ))}
              <li className={styles.li}>
                <Link href={donateLink}>
                  <a className={styles.a}>Donate</a>
                </Link>
              </li>
            </ul>
          </div>
        )}
      </header>
    );
  }
}
