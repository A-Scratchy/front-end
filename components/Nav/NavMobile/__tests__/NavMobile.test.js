import React from 'react';
import { shallow } from 'enzyme';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import CloseButton from 'components/_common_/CloseButton/CloseButton';
import NavMobile from '../NavMobile';

describe('NavMobile', () => {
  it('should render', () => createShallowSnapshotTest(<NavMobile />));

  it('dropdown is not visible without hamburger cutton click', () => {
    const wrapper = shallow(<NavMobile />);

    expect(wrapper.find('ul')).not.toExist();
  });

  it('dropdown is visible with hamburger button click', () => {
    const wrapper = shallow(<NavMobile />);

    wrapper.find('.hamburgerButtonWrapper').simulate('click');

    expect(wrapper.find('ul')).toExist();
  });

  it('visible dropdown should have 10 links', () => {
    const wrapper = shallow(<NavMobile />);

    wrapper.find('.hamburgerButtonWrapper').simulate('click');

    expect(wrapper.find('ul').find('a')).toHaveLength(10);
  });

  it('visible dropdown dissappears on close button click', () => {
    const wrapper = shallow(<NavMobile />);

    wrapper.find('.hamburgerButtonWrapper').simulate('click');
    expect(wrapper.find('ul')).toExist();
    wrapper.find(CloseButton).simulate('click');

    expect(wrapper.find('ul')).not.toExist();
  });
});
