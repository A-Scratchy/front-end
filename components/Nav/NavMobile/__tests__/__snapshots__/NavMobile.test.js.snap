// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMobile should render 1`] = `
<header
  className="NavMobile"
>
  <Link
    href="/"
  >
    <img
      alt="Operation Code Logo"
      className="logo"
      src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-blue-logo.png"
    />
  </Link>
  <button
    className="hamburgerButtonWrapper"
    name="dropdown"
    onClick={[Function]}
    type="button"
  >
    <Svg
      className="hamburgerIcon"
    />
  </button>
</header>
`;
